### ISSUE TITLE
Improve handling of mega objects

### ISSUE THREAD

**Issue by tillkruss** [WordPress/performance#105]
Extracted from #36.

The `alloptions` cache key can be several hundred megabytes and could use better handling. Some 3rd party plugins handle this by splitting it up. Can core do the same?

Instead of focussing only of the `alloptions` keys, maybe cache keys can be designated as _MEGA_ and object cache implementations can handle them as they please.

**Comment by peterwilsoncc** [comment]
There is some related discussion on https://core.trac.wordpress.org/ticket/31245 -- it's quite long and covers a range of subjects around and including alloptions.

**Comment by westonruter** [comment]
I don't think this is something which could conceivably be implemented as a performance feature in plugin form. So I think a Trac ticket would be needed, if one doesn't exist yet, to carry this forward.
